OneCommunityGlobal/HighestGoodNetworkApp
# Shengwei_HotFix: Page render issue in Timelog

# Description
Hotfix blank page in the main site due to the user's timelog entry is linked to a project from which the user has been removed.

<img width="1093" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/0652a37e-2034-4c06-bcc4-c50caae4e710">

Fix rendering issue in Timelog
- Ensure that the page renders all time entries regradless of projects the user is assigned to
- Add a warning to time entries associated with projects that the user no longer has access to
- Prevent time entries from being updated if associated with projects that the user no longer has access to.

## Related PRS (if any):
Backend [PR898](https://github.com/OneCommunityGlobal/HGNRest/pull/898)

## Main changes explained:
- Update file src/components/Timelog/TimeEntry.jsx for including new logic for time entries rendering.
<img width="888" alt="hotfix_main_time_entry_issue" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/aab5d1b7-3e11-4e86-82a9-9e2e4d4c0ec9">


## How to test:
1. check into current branch and checkout backend [PR898](https://github.com/OneCommunityGlobal/HGNRest/pull/898)
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard -> timelog
6. log a time entry under a specific project
7. remove the user from the project 
8. reload the timelog page

## Screenshots or videos of changes:

## Note:
Include the information the reviewers need to know.
